tests/namespace.go

```diff
@@ -0,0 +1,277 @@
+/*
+ * This file is part of the KubeVirt project
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ * Copyright 2022 Red Hat, Inc.
+ *
+ */
+
+package tests
+
+import (
+	"context"
+	"fmt"
+	"time"
+
+	. "github.com/onsi/ginkgo/v2"
+	. "github.com/onsi/gomega"
+
+	k8sv1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime/schema"
+	"k8s.io/apimachinery/pkg/types"
+
+	v1 "kubevirt.io/api/core/v1"
+	"kubevirt.io/client-go/kubecli"
+	"kubevirt.io/kubevirt/pkg/controller"
+
+	"kubevirt.io/kubevirt/tests/flags"
+	"kubevirt.io/kubevirt/tests/framework/cleanup"
+	"kubevirt.io/kubevirt/tests/libnet"
+	"kubevirt.io/kubevirt/tests/util"
+)
+
+// NamespaceTestAlternative is used to test controller-namespace independently.
+var NamespaceTestAlternative = "kubevirt-test-alternative"
+
+// NamespaceTestOperator is used to test if namespaces can still be deleted when kubevirt is uninstalled
+var NamespaceTestOperator = "kubevirt-test-operator"
+
+var TestNamespaces = []string{util.NamespaceTestDefault, NamespaceTestAlternative, NamespaceTestOperator}
+
+func cleanNamespaces() {
```

Comment:
I would say this is more of testsuite/internal stuff. WDYT?

Reply:
Sounds like another `utils` generic naming to me.
I agree it does a lot of other stuff, but the best relation I found so far is to the namespace. Cleaning all its content has some relation to "namespace".

Ideally, in my view, we should just create a namespace per test and just get rid of these details (which are fragile and fail to scale with the increase of new objects types used in tests). But that is another topic for e2e test cleanup.

Reply:
I wouldn't say so. My point is that one would be now searching for useful functions that are manipulating namespaces here. But imo these are only internal and should not be used in the tests. I understand you wanted to do minimal work here to enable the ARM tests but let's revisit this.

Reply:
Now I get it, thanks for clarifying this.
I guess it is the same answer posted [here](https://github.com/kubevirt/kubevirt/pull/7625#discussion_r860636891).